state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -331,6 +368,19 @@ pub mod pallet {
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
+		/// New swap has been requested
+		SwapRequested {
+			swap_request_id: SwapRequestId,
+			input_asset: Asset,
+			input_amount: AssetAmount, // includes broker fee
+			output_asset: Asset,
+			broker_fee: AssetAmount,
+			origin: SwapOrigin,
+			request_type: RequestTypeEncoded,
```

Comment:
I'm not sure about this. `SwapRequest` also contains state that changes throughout the request's lifecycle, and I the event should only expose the parameters rather than its dynamic state.